spring-integration-core/src/main/java/org/springframework/integration/util/MessagingMethodInvokerHelper.java

```diff
@@ -979,7 +979,7 @@ private void setExclusiveTargetParameterType(TypeDescriptor targetParameterType,
 		}
 	}
 
-	private static class ParametersWrapper {
+	public static class ParametersWrapper {
```

Comment:
I am not so sure we should make it top level; it's pretty specialized.

Also, I just noticed that `HandlerMethod.getExpression()` is no longer needed since you changed the code to use the field directly.

Doesn't IDEA warn you of unused methods - time to switch to STS ? 😄